crates/runtime/src/lib.rs

```diff
@@ -1565,6 +1593,149 @@ impl Runtime {
         Ok(())
     }
 
+    async fn apply_model_diff(&self, current_app: &Arc<App>, new_app: &Arc<App>) {
```

Comment:
Except for `apply_view_diff`, this is just refactored out of [start_pods_watcher](https://github.com/spiceai/spiceai/pull/3428/files#diff-c6d4f276260e77036ce0adaf946b9ee6e55b1af1640342035121d4b254090df8R1739).